openmetadata-ui/src/main/resources/ui/src/pages/AddAlertPage/AddAlertPage.tsx

```diff
@@ -0,0 +1,652 @@
+        filteringRules: requestFilteringRules,
+      });
+    } catch {
+      showErrorToast(
+        t('message.entity-fetch-error', { entity: t('label.alert') }),
+        fqn
+      );
+    } finally {
+      setLoadingCount((count) => count - 1);
+    }
+  };
+
+  useEffect(() => {
+    if (fqn) {
+      fetchAlert();
+    }
+  }, [fqn]);
+
+  const fetchFunctions = async () => {
+    try {
+      setLoadingCount((count) => count + 1);
+      const functions = await getFilterFunctions();
+      const entityFunctions = await getEntityFilterFunctions();
+
+      setFilterFunctions(functions);
+      setEntityFunctions(entityFunctions);
+    } finally {
+      setLoadingCount((count) => count - 1);
+    }
+  };
+
+  const fetchDefaultTriggerConfig = async () => {
+    setLoadingCount((count) => count + 1);
+    const triggers = await getDefaultTriggerConfigs();
+
+    setDefaultTriggers(
+      triggers.map((trigger) => ({
+        ...trigger,
+        entities: trigger.entities?.sort(),
+      }))
+    );
+    setLoadingCount((count) => count - 1);
+  };
+
+  useEffect(() => {
+    fetchFunctions();
+    fetchDefaultTriggerConfig();
+  }, []);
+
+  const handleSave = async (data: Alerts) => {
+    const { filteringRules, alertActions } = data;
+
+    const requestFilteringRules = filteringRules?.map((curr) => ({
+      ...curr,
+      condition: `${curr.name}(${map(
+        curr.condition,
+        (v: string) => `'${v}'`
+      )?.join(', ')})`,
+    }));
+
+    try {
+      const actions: AlertAction[] =
+        alertActions?.map(
+          (action) =>
+            ({
+              ...action,
+              enabled: true,
+            } as unknown as AlertAction)
+        ) ?? [];
+
+      const promises = actions.map((action) => createAlertAction(action));
+
+      const responses = await Promise.allSettled(promises);
+
+      const requestAlertActions: EntityReference[] = responses.map((res) => {
+        if (res.status === PROMISE_STATE.REJECTED) {
+          throw res.reason;
+        }
+
+        return {
+          id: res.status === PROMISE_STATE.FULFILLED ? res.value.id ?? '' : '',
+          type: 'alertAction',
+        };
+      });
+
+      try {
+        await createAlert({
+          ...data,
+          filteringRules: requestFilteringRules,
+          alertActions: requestAlertActions,
+        });
+
+        showErrorToast(
+          t('server.create-entity-success', { entity: t('alert-plural') })
+        );
+        history.push(
+          getSettingPath(
+            GlobalSettingsMenuCategory.COLLABORATION,
+            GlobalSettingOptions.ALERTS
+          )
+        );
+      } catch (error) {
+        showErrorToast(
+          t('server.entity-creation-error', {
+            entity: t('label.alert-plural'),
+          }),
+          (error as AxiosError).message
+        );
+        showErrorToast(
+          `Unable to create alert.`,
+          (error as AxiosError).message
+        );
```

Comment:
is it intentional to throw two error notifications? also localization is missing

Reply:
Thanks for catching